universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java

```diff
@@ -0,0 +1,100 @@
+package views.admin.questions;
+
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.br;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.label;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Optional;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.question.QuestionDefinition;
+import services.question.QuestionType;
+import services.question.TranslationNotFoundException;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+
+public final class QuestionEditView extends BaseHtmlView {
+  private final BaseHtmlLayout layout;
+
+  @Inject
+  public QuestionEditView(BaseHtmlLayout layout) {
+    this.layout = layout;
+  }
+
+  public Content render(Request request, Optional<QuestionDefinition> question) {
+    String headerText = question.isPresent() ? "Edit Question" : "New Question";
+    return layout.htmlContent(
+        body(
+            renderHeader(headerText),
+            buildQuestionForm(question.orElse(null)).with(makeCsrfTokenInputTag(request))));
```

Comment:
Suggestion (non-blocking): I think it would be more readable to pass down optional and check `isPresent` below instead of `null`.

Reply:
+1 please do not intentionally pass `null` to methods or constructors if ever avoidable